### ISSUE TITLE
error about batch

### ISSUE THREAD

**Issue by zhang-wen** [clab/dynet#187]
no error when i train my model on GPU, but if i train it on CPU, there is an error here:

Eigen::Map<Eigen::Matrix<float, -1, -1> > dynet::Tensor::operator*(): Assertion `d.batch_elems() == 1' failed.

i debug, error happens for calling `backward`

**Comment by neubig** [comment]
Please give us more information and a minimum reproducible example.

**Comment by neubig** [comment]
No word on this, so I'm closing, but feel free to re-open if you can create an example.

**Comment by vered1986** [comment]
I've had the exact same problem. and I didn't manage to reproduce it with a minimal example, but eventually I solved it. My problem was that somewhere along the computation I tried accessing row/columns/cells in expression objects like I do with numpy matrices, and it returned unexpected outputs. I ended up reshaping the matrices to vectors and accessing cells linearly using row-major order. I'm not sure whether I should open a separate bug on the indexing or not.
